### DISCUSSION TITLE
How to update the Teleport license for a Kubernetes deployment

### DISCUSSION THREAD

**Discussion by blorby** [gravitational/teleport#36243] [Show and tell]
1. make sure you are on the right context cluster by checking the teleport namespace `kubectl get namespaces`
2. verify the new license in your path and valid: `openssl x509 -text -in license.pem | grep "Not After"`
3. delete the current license: `kubectl -n teleport delete secret license`
4. create a new secret: `kubectl -n teleport create secret generic license --from-file=license.pem`
5. restart (delete) the teleport-auth pods: `kubectl -n teleport get pods | grep auth | cut -d' ' -f 1 | xargs kubectl  -n teleport delete pod
`

**Comment by webvictim** [comment]
Thanks for this write-up!

**Reply by blorby** [reply]
can you add it to the documentation?
also, I've started writing an upgrade process, you already have something like that?
